receiver/azureblobrehydrationreceiver/receiver.go

```diff
@@ -109,170 +124,202 @@ func newRehydrationReceiver(id component.ID, logger *zap.Logger, cfg *Config) (*
 		id:              id,
 		cfg:             cfg,
 		azureClient:     azureClient,
-		doneChan:        make(chan struct{}),
 		checkpointStore: rehydration.NewNopStorage(),
 		startingTime:    startingTime,
 		endingTime:      endingTime,
-		ctx:             ctx,
-		cancelFunc:      cancel,
+		blobChan:        make(chan []*azureblob.BlobInfo, blobChanSize),
+		errChan:         make(chan error),
+		doneChan:        make(chan struct{}),
+		mut:             &sync.Mutex{},
+		wg:              &sync.WaitGroup{},
 	}, nil
 }
 
 // Start starts the rehydration receiver
 func (r *rehydrationReceiver) Start(ctx context.Context, host component.Host) error {
+	r.logAnyDeprecationWarnings()
 
 	if r.cfg.StorageID != nil {
 		checkpointStore, err := rehydration.NewCheckpointStorage(ctx, host, *r.cfg.StorageID, r.id, r.supportedTelemetry)
 		if err != nil {
 			return fmt.Errorf("NewCheckpointStorage: %w", err)
 		}
-
 		r.checkpointStore = checkpointStore
 	}
 
-	r.started = true
-	go r.scrape()
+	cancelCtx, cancel := context.WithCancel(ctx)
+	r.cancelFunc = cancel
+
+	go r.streamRehydrateBlobs(cancelCtx)
 	return nil
 }
 
+func (r *rehydrationReceiver) logAnyDeprecationWarnings() {
+	if r.cfg.PollInterval != 0 {
+		r.logger.Warn("poll_interval is no longer recognized and will be removed in a future release. batch_size/page_size should be used instead")
+	}
+
+	if r.cfg.PollTimeout != 0 {
+		r.logger.Warn("poll_timeout is no longer recognized and will be removed in a future release. batch_size/page_size should be used instead")
+	}
+}
+
 // Shutdown shuts down the rehydration receiver
 func (r *rehydrationReceiver) Shutdown(ctx context.Context) error {
-	r.cancelFunc(errors.New("shutdown"))
-	var err error
-
-	// If we have called started then close and wait for goroutine to finish
-	if r.started {
-		select {
-		case <-ctx.Done():
-			err = ctx.Err()
-		case <-r.doneChan:
-		}
+	if r.cancelFunc != nil {
+		r.cancelFunc()
 	}
 
-	err = errors.Join(err, r.checkpointStore.Close(ctx))
+	shutdownCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
 
-	return err
-}
+	// signal shutdown intent
+	close(r.doneChan)
 
-// emptyPollLimit is the number of consecutive empty polling cycles that can
-// occur before we stop polling.
-const emptyPollLimit = 3
+	// wait for any in-progress operations to finish
+	done := make(chan struct{})
+	go func() {
+		r.wg.Wait()
+		close(done)
+	}()
 
-// scrape scrapes the Azure api on interval
-func (r *rehydrationReceiver) scrape() {
-	defer close(r.doneChan)
-	ticker := time.NewTicker(r.cfg.PollInterval)
-	defer ticker.Stop()
+	select {
+	case <-done:
+	case <-shutdownCtx.Done():
+		return fmt.Errorf("shutdown timeout: %w", shutdownCtx.Err())
+	}
 
-	var marker *string
+	var errs error
+	if err := r.makeCheckpoint(shutdownCtx); err != nil {
+		r.logger.Error("Error while saving checkpoint", zap.Error(err))
+		err = errors.Join(err, err)
+	}
+
+	return errors.Join(errs, r.checkpointStore.Close(shutdownCtx))
```

Comment:
Generally try to wrap errors, and since we're combining the errors from these 2 calls I don't think we need to log the error from the first one since it'll get reported once shutdown is done. Thoughts?
```suggestion
	var errs error
	if err := r.makeCheckpoint(shutdownCtx); err != nil {
		errs = errors.Join(errs, fmt.Errorf("make checkpoint: %w", err))
	}
	if err := r.checkpointStore.Close(shutdownCtx); err != nil {
	         errs = errors.Join(errs, fmt.Errorf("close checkpoint store: %w", err))
	}
        return errs
```

Reply:
If you decide to checkpoint after every blob though, I don't think we need to worry about checkpointing in shutdown, so the first error block could get removed, but that's if the checkpointing logic changes